### DISCUSSION TITLE
How to login to MicroStrategy Library API in Postman via SAML / OKTA Auth

### DISCUSSION THREAD

**Discussion by kkadu** [MicroStrategy/rest-api-playground#11] [Q&A]
We have enabled SAML / OKTA for enterprise application and I tried to set auth mode as 2 when setting up the environment in postman for MSTR but I am not able to login. Has anyone tried with this type of authentication?

**Comment by xiakunhou** [comment]
@kkadu Can you log in from a workstation, library, or web? 
Postman should now work well with saml or oauth which require browser redirect.
If you want to login through postman, you can create a API token for the user and login with the API token.

**Reply by kkadu** [reply]
yes, I can login to library from browser with SAML / OKTA single-signon but not to the API endpoint. I did check on the supported auth mode for the rest api (below link) but it seems that it is not supported currently.
https://github.com/MicroStrategy/mstrio/issues/20 
To create the identity token (/api/auth/identityToken), I am not able to authenticate the user login (/api/auth/login) with login mode = 2

**Comment by rd152343** [comment]
@kkadu 
Can you please let me know if you are able to overcome this issue? We are facing the same issue but no luck
